test/functional/wallet_v3_txs.py

```diff
@@ -0,0 +1,560 @@
+
+    @cleanup
+    def send_v3(self):
+        self.log.info("Test setting version to 3 with `send`")
+        self.generate(self.nodes[2], 1)
+
+        outputs = {self.alice.getnewaddress() : 10}
+        tx_hex = self.charlie.send(outputs=outputs, add_to_wallet=False, version=3)["hex"]
+        assert_equal(self.charlie.decoderawtransaction(tx_hex)["version"], 3)
+
+    @cleanup
+    def sendall_v3(self):
+        self.log.info("Test setting version to 3 with sendall")
+        self.generate(self.nodes[2], 1)
+
+        tx_hex = self.charlie.sendall(recipients=[self.alice.getnewaddress()], version=3, add_to_wallet=False)["hex"]
+        assert_equal(self.charlie.decoderawtransaction(tx_hex)["version"], 3)
+
+    @cleanup
+    def sendall_with_unconfirmed_v3(self):
+        self.log.info("Test setting version to 3 with sendall + unconfirmed inputs")
+
+        inputs=[]
+
+        # confirmed v2 utxos
+        outputs = {self.charlie.getnewaddress() : 2.0}
+        confirmed_v2 = self.send_tx(self.charlie, inputs, outputs, 2)
+
+        # confirmed v3 utxos
+        outputs = {self.charlie.getnewaddress() : 2.0}
+        confirmed_v3 = self.send_tx(self.charlie, inputs, outputs, 3)
+
+        self.generate(self.nodes[2], 1)
+
+        # unconfirmed v2 utxos
+        outputs = {self.charlie.getnewaddress() : 2.0}
+        unconfirmed_v2 = self.send_tx(self.charlie, inputs, outputs, 2)
+
+        # unconfirmed v3 utxos
+        outputs = {self.charlie.getnewaddress() : 2.0}
+        unconfirmed_v3 = self.send_tx(self.charlie, inputs, outputs, 3)
+
+        # Test that the only unconfirmed inputs this v3 tx spends are v3
+        tx_hex = self.charlie.sendall([self.bob.getnewaddress()], version=3, add_to_wallet=False)["hex"]
+
+        decoded_tx = self.charlie.decoderawtransaction(tx_hex)
+        decoded_vin_txids = [txin["txid"] for txin in decoded_tx["vin"]]
+
+        assert_equal(decoded_tx["version"], 3)
+
+        assert confirmed_v3 in decoded_vin_txids
+        assert confirmed_v2 in decoded_vin_txids
+        assert unconfirmed_v3 in decoded_vin_txids
+        assert unconfirmed_v2 not in decoded_vin_txids
+
+        # Test that the only unconfirmed inputs this v2 tx spends are v2
+        tx_hex = self.charlie.sendall([self.bob.getnewaddress()], version=2, add_to_wallet=False)["hex"]
+
+        decoded_tx = self.charlie.decoderawtransaction(tx_hex)
+        decoded_vin_txids = [txin["txid"] for txin in decoded_tx["vin"]]
+
+        assert_equal(decoded_tx["version"], 2)
+
+        assert confirmed_v3 in decoded_vin_txids
+        assert confirmed_v2 in decoded_vin_txids
+        assert unconfirmed_v2 in decoded_vin_txids
+        assert unconfirmed_v3 not in decoded_vin_txids
+
+    @cleanup
+    def walletcreatefundedpsbt_v3(self):
+        self.log.info("Test setting version to 3 with walletcreatefundedpsbt")
+        self.log.info("Test setting version to 3 with createpsbt")
```

Comment:
In 09c9f828250e3840d7557374ff568204305e2679 "test: add truc wallet tests"

This log is not supposed to be here.

Reply:
I've removed it.